packages/expo-file-system/ios/EXFileSystem/EXFileSystem.m

```diff
@@ -961,8 +961,21 @@ - (void)sendEventWithName:(NSString *)eventName body:(id)body
   }
 }
 
-- (NSDictionary *)documentFileSystemAttributes {
-  return [[NSFileManager defaultManager] attributesOfFileSystemForPath:_documentDirectory error:nil];
+- (NSDictionary<NSURLResourceKey, id> *)documentFileResourcesForKeys:(NSArray<NSURLResourceKey> *)keys
+{
+  if (!keys.count) {
+    return @{};
+  }
+
+  NSURL *documentDirectoryUrl = [NSURL fileURLWithPath:_documentDirectory];
+  NSError *error = nil;
+  NSDictionary *results = [documentDirectoryUrl resourceValuesForKeys:keys error:&error];
+  if (!results) {
+      NSLog(@"Error retrieving resource keys: %@\n%@", [error localizedDescription], [error userInfo]);
```

Comment:
it might be useful to propagate this error message that we receive back up to the `getFreeDiskStorageAsyncWithResolver` method, so that we can reject the promise and include the reason for the error in that rejection

Reply:
@cruzach Just pushed a new commit with the suggested changes.